### ISSUE TITLE
torch.sparse_coo_tensor argname quirks

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#109017]
### 🐛 Describe the bug

`torch.sparse_coo_tensor(indices, values, size=None)` but in SciPy `scipy.sparse.coo_array((data, (i, j)), [shape=(M, N)])`

Note that vals/indices order is flipped + PyTorch uses `size` instead of `shape`. There are similar quirks for `torch.sparse_csr_tensor`.

I would suggest that PyTorch also has alias argname `shape=` for `size=`, as it matches more with NumPy/SciPy, e.g. https://numpy.org/doc/stable/reference/generated/numpy.zeros.html vs https://pytorch.org/docs/stable/generated/torch.zeros.html (the argname `size=` is likely the remnant of LuaTorch, but now `.shape` has mostly replaced `.size()`)

### Versions

N/A

**Comment by vadimkantorov** [comment]
Currently getting `.values()` on a not-coalesced COO tensor gives: `RuntimeError: Cannot get values on an uncoalesced tensor, please call .coalesce() first`

Is it a meaningful restriction? One could think that if we care about the inner elements (values/indices) we already know what we are doing, so actually letting access them as is might be okay. Currently we need to instead use private `_values()` - e.g. for constructing a scipy.sparse.coo_matrix from an equivalent torch.sparse_coo_tensor

**Comment by nikitaved** [comment]
values() is a differentiable function. And as such it can only be run on a coalesced inputs. Otherwise, the inflowing grad needs to get accumulated and averaged over the groups of repeated indicies. This is quite prohibitive performance-wise, hence the requirements for being coalesced. The grad propagation is much easier in this case.

**Comment by vadimkantorov** [comment]
Hmm, understood.

Maybe alternative could be `.values(force=True)` similar to `numpy()` or maybe instead `_values()` could become more public/documented/renamed - as it's needed for various interop scenarios

**Comment by nikitaved** [comment]
I agree. I do not like _values as it hits at being non-public. I'd rather have something like values_view or something. This way it is also hinted that the user should be careful when doing autograd similar to other view ops...

**Comment by vadimkantorov** [comment]
(original usecase was just wanting to convert sparse_coo/sparse_csr to numpy for using their existing linear system solvers)
